packages/playwright-core/src/server/frames.ts

```diff
@@ -965,20 +972,25 @@ export class Frame extends SdkObject {
   async retryWithProgress<R>(
     progress: Progress,
     selector: string,
-    options: types.StrictOptions & types.TimeoutOptions,
-    action: (frame: Frame, info: SelectorInfo, continuePolling: symbol) => Promise<R | symbol>,
+    options: types.StrictOptions & types.TimeoutOptions & { omitAttached?: boolean },
+    action: (selector: SelectorInFrame | null, continuePolling: symbol) => Promise<R | symbol>,
     scope?: dom.ElementHandle): Promise<R> {
+    progress.log(`waiting for selector "${selector}"`);
```

Comment:
Now that we have "waiting for frame <selector>" and "waiting for selector <selector>" in the last frame, perhaps we do not need this one? It contains strange things like "control=" that users never entered :)

Reply:
Sounds good, I might need to rebase expectations for this one, let me check.